pageserver/src/tenant/layer_map.rs

```diff
@@ -341,17 +545,7 @@ impl LayerMap {
 
     /// Return all L0 delta layers
     pub fn get_level0_deltas(&self) -> Result<Vec<Arc<dyn Layer>>> {
-        let mut deltas = Vec::new();
-        for l in self.historic_layers.iter() {
-            if !l.is_incremental() {
-                continue;
-            }
-            if l.get_key_range() != (Key::MIN..Key::MAX) {
-                continue;
-            }
-            deltas.push(Arc::clone(l));
-        }
-        Ok(deltas)
+        Ok(self.l0_layers.clone())
```

Comment:
The old code only returned delta layers, i.e. layers with `is_incremental()`. The new code returns all layers in `l0_layers`, which also includes layers with `!is_incremental()`. I don't think we currently ever create non-incremental layers with key-range MIN..MAX, so doesn't make a difference in practice, but it feels like it could cause confusion and bugs in the future.

Reply:
Well, my assumption really was that L2 layer can only be incremental. In principle I can imagine situation when we want to enforce creation of the image layer instead of delta layer (bulk load of data is good candidate for it).
But then the question arrive - which layers are considered to belong to level 0? What we have no is not a classical LSM hierarchy. L0 layers are just intermediate layers and input for the compaction (from my point of view, may be you disagree).
In this case there is no sense to treat image layers (even if them are created from the very beginning as L0 layers).
Please notice that the only placed where `get_level0_deltas` is `compact_level0`. So this fuction actually returns input for the compaction.

May be should just rename `l0_layers` to `l0_delta_layer` ?  Or do you prefer some other solution.

Reply:
> May be should just rename l0_layers to l0_delta_layer ? Or do you prefer some other solution.

Works for me